File: medium/71.简化路径/71.简化路径.py
#
# @lc app=leetcode.cn id=71 lang=python3
#
# [71] 简化路径
#


class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = ['']
        arr = path.split('/')
        for s in arr:
            if s == '' or s == '.':
                continue
            elif s == '..':
                stack.pop()
                if len(stack) == 0:
                    stack.append('')
            else:
                stack.append(s)
        return '/' if len(stack) == 1 else '/'.join(stack)

File: medium/71.简化路径/71.简化路径.js
/*
 * @lc app=leetcode.cn id=71 lang=javascript
 *
 * [71] 简化路径
 */
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
  const stack = ['']
  const arr = path.split('/')
  for (const str of arr) {
    if (str === '' || str === '.') {
      continue
    } else if (str === '..') {
      stack.pop()
      if (!stack.length) {
        stack.push('')
      }
    } else {
      stack.push(str)
    }
  }
  return stack.length === 1 ? '/' : stack.join('/')
}
